stack.yaml

```diff
@@ -18,7 +18,9 @@
 resolver: lts-11.3
 
 nix:
-  packages : [ zlib ]
+  packages:
+    - z3
```

Comment:
Will this install at least the 4.7.0 version or whatever is needed for our project?

Reply:
It will use exactly the version of Z3 is in the user's Nixpkgs expression. Nix packages are pure functions from dependencies to build artifacts; all package versions are _exact_ to ensure reproducibility. If you would prefer, I can provide a Nix expression in the Kore repository which would provide an appropriate version.